### PR TITLE
Set catalog placeholder image background based on color hash

### PR DESCRIPTION
**Pull request by marians**
The idea here is to give a unique color background to every catalog icon that does not have an image.

image

### CODE REVIEW

**Comment by marians** [comment]
@giantswarm/chapter-front-end Any idea how to get the rounded corners right? There is white background appearing in the corners.

image

**Comment by oponder** [comment]
The way I do this fallback is a bit of a hack perhaps, because I found it as a more straightforward way of handling it actually than the onError way we did in the AppListing. If we can get it to work nicely here, we can apply it in the app listing too eventually and remove some code to keep track of which images failed to load.

Anyways, I think what we might be seeing in the top left corner is the missing image icon, peaking through beneath the element that I am adding over it.

Not sure why the other corners aren't perfectly transparent though.. 🤔 

For illustration purposes, I disabled some of the ":before" styles:

Screenshot 2020-06-23 at 3 19 27 PM

There might be a way to adjust the position or look of that missing image icon to at least clean up the top left border.

**Comment by ubergesundheit** [comment]
Checked this out locally and it does not work in either Firefox 77.0.1 nor Chromium  80.0.3987.149 for me:

I suspect setting it needs the `background-color` also being set in the `:before`-style of `Image`.

The white top left corner does not appear in Firefox, I guess oponders suggestion is right, it should be possible to attach an event listener to `onError` of the image and then replace the element with a div

Left Chromium, right Firefox:
image

**Comment by marians** [comment]
This is going beyond me already. Anyone to take this over?

**Comment by ubergesundheit** [comment]
I can do it

File: src/utils/cachingColorHash.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import ColorHash from 'color-hash';
+
+class CachingColorHash {
+  private cache: { [k: string]: string } = {};
```
Comment (axbarsan) [inline src/utils/cachingColorHash.ts:4]:
```suggestion
  private cache: Record<string, string> = {};
```

**Pull request merged**